tests/foreman/ui/test_discoveredhost.py

```diff
@@ -229,25 +173,43 @@ def test_positive_auto_provision_host_with_rule(
 
     :Setup: Host should already be discovered
 
-    :expectedresults: Host should reboot and provision
+    :expectedresults: Host should be successfully provisioned
 
     :BZ: 1665471, 1731112
 
     :CaseImportance: High
     """
+    sat = module_discovery_sat.sat
+    pxeless_discovery_host.power_control(ensure=False)
+    mac = pxeless_discovery_host._broker_args['provisioning_nic_mac_addr']
+    wait_for(
+        lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+        timeout=240,
+        delay=20,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+
+    discovered_host_name = discovered_host.name
+    domain_name = provisioning_hostgroup.domain.read().name
+    host_name = f'{discovered_host_name}.{domain_name}'
+
     host_ip = gen_ipaddr()
```

Comment:
As you'll be running this on provisioning setup so shouln't we pick IP from related subnet here instead?

Reply:
Yeah you are right I think the assertion itself won't be valid here